Crypto/ECC/Ed25519.hs

```diff
@@ -0,0 +1,298 @@
+-- |
+-- Module      : Crypto.ECC.Ed25519
```

Comment:
In some other related works, I tried to distinguish Ed25519 (the signature system) from the curve (Edwards25519). do you think it might be worthwhile to do that here at the module level ?

Reply:
Yes I was entirely focused on the prefix `Crypto.ECC` vs `Crypto.PubKey` but renaming the module `Edwards25519` will make it even clearer. So I'll change it, as well as the curve name for consistency.